featuretools/io/sql.py

```diff
@@ -0,0 +1,100 @@
+import pandas as pd
+import sqlalchemy as sa
+
+import featuretools as ft
+
+
+def from_sql(id, connection=None, connection_string=None, tables=None, index={},
+             time_index={}, secondary_time_index={}, encoding={}, variable_types={}):
+    if connection is None:
+        if connection_string is None:
+            raise ValueError('connection or connection_string is required')
+        else:
+            connection = sa.create_engine(connection_string, echo=False).connect()
+
+    loader = EntitySetLoaderFromSQL(connection=connection, id=id, index=index, time_index=time_index,
+                                    secondary_time_index=secondary_time_index, encoding=encoding,
+                                    variable_types=variable_types)
+    loader.load_entityset(tables)
+    return loader.es
+
+
+def _find_pk_name(table):
+    column_names = table.columns.keys()
+    pks = [name for name in column_names if table.columns[name].primary_key]
+    if len(pks) > 1:
+        raise RuntimeError('Composite primary key detected')
+    elif len(pks) == 1:
+        return pks[0]
+    else:
+        return None
+
+
+class EntitySetLoaderFromSQL:
+    def __init__(self, id, connection, **passthrough_args):
+        self.connection = connection
+        self.metadata = sa.MetaData(connection)
+        self.metadata.reflect(bind=self.connection.engine)
+        self.passthrough_args = passthrough_args
+        self.table_names = self._get_table_names()
+        self.es = ft.EntitySet(id=id)
+
+    def load_entityset(self, tables):
+        self.load_tables(tables)
+        self.load_relationships()
+
+    def _get_table_names(self):
+        return list(self.metadata.tables.keys())
+
+    def _get_args_for_table(self, table_name):
+        return {k: v.get(table_name) for k, v in self.passthrough_args.items()}
+
+    def _load_table(self, table_name):
+        table = sa.Table(table_name, self.metadata, autoload=True)
+        df = pd.read_sql(table.select(), table.metadata.bind)
+
+        table_args = self._get_args_for_table(table_name)
+        print("table_args:", table_args)
+        if 'index' not in table_args or table_args['index'] is None:
+            table_args['index'] = _find_pk_name(table)
+
+        return self.es.entity_from_dataframe(
+            entity_id=table_name,
+            dataframe=df,
+            **table_args
+        )
+
+    def _column_fk_list(self, table, column_name):
+        fks = table.columns[column_name].foreign_keys
+        return list(filter(lambda fk: fk, fks))
+
+    def _load_table_fks(self, table_name):
+        table = sa.Table(table_name, self.metadata, autoload=True)
+        column_names = table.columns.keys()
+        fks = {column_name: self._column_fk_list(table, column_name) for column_name in column_names}
+        return fks
+
+    def load_relationships(self):
```

Comment:
do we handle the case where you only include one of the tables that's part of a relationship? 

say we have table A which is a child of table B. but a user only want to load in table A. do we handle the situation where we try to add the relationship for A to B, but B doesn't exist?

Reply:
I believe this testcase handles what you're thinking about:
https://github.com/Featuretools/featuretools/pull/305/files#diff-9d3216b92b716cc50a8429e1d3350b4bR44

specifically, since we load widgets & factories, but don't load customers, only the widgets <> factories relationship is loaded. 

Let me know if there's another case we should test.

Reply:
here is the case

```
def test_load_only_child_in_relationship(sqlite):
    es = ft.io.from_sql(id="widget_limited_tables", connection=sqlite, tables=["customers_widgets"])
    assert(len(es.relationships) == 0)
    assert(len(es.entities) == 1)
```

when I run this I get the following error 
```
*** KeyError: 'Entity customers does not exist in widget_limited_tables'
```